### PR TITLE
Uploading images

### PR DESCRIPTION
**Pull request by azayneeva**
This pr relates to issue #125 
1. For uploading images I used 2 modules: react-dropzone and cloudinary, as was suggested in this article (you will need to do npm install for package.json in client folder): https://css-tricks.com/image-upload-manipulation-react/ 
The only thing that didn't work from this article was this code:
```
handleImageUpload(file) {
    let upload = request.post(CLOUDINARY_UPLOAD_URL)
                        .field('upload_preset', CLOUDINARY_UPLOAD_PRESET)
                        .field('file', file);
```
But after some research I changed the code to this one:
```
  handleImageUpload(file) {
    const cloudinaryForm = new FormData();
    cloudinaryForm.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);
    cloudinaryForm.append('file', file);

    fetch(CLOUDINARY_UPLOAD_URL, {
      method: 'POST',
      body: cloudinaryForm,
      headers: {
        Accept: 'application/json',
      },
 ...
```
And everything started working :)
2. Then I updated our database schema to accept cb logo url, updated backend routes as well
3. The problem was with pdfmake module, it doesn't accept external urls for images, only 64base or local files. For this reason I had to find the way to encode url into base64. I installed module called node-base64-image (you will need to do npm install for the main package.json). It has this function that encodes url:
```
base64.encode(url, options, (err, image) => {})
```
I converted it into this one in pdfgenerator.js (without string: true it didn't work):
```
base64.encode(url, { string: true }, (err, image) => {})
```
And seems everything is working now :)
But this pr is huge again :(

### CODE REVIEW

**Comment by finnhodgkin** [comment]
Seems to work! The UX is a little strange - it feels like it's already uploaded when the picture first arrives but then if you go back it hasn't because submit hasn't been hit. I think there should be more indication that changes only happen once submit is clicked.

I also think that when an image is uploaded the drag and drop area should be replaced with the image, and a red x should appear in the upper right corner of the image, allowing you to cancel the image selection and bring back the drag and drop area.

(still reviewing the code)

File: react-backend/functions/pdfgenerator.js [hunk 1]
```diff
@@ -22,46 +86,32 @@ module.exports = QRcodeBase64Url => {
...
-      result = Buffer.concat(chunks);
-      resolve(result.toString('base64'));
-      reject("Couldn't generate PDF");
-    });
-    doc.end();
+  doc.on('data', chunk => {
+    chunks.push(chunk);
+  });
+  doc.on('end', () => {
+    result = Buffer.concat(chunks);
```
Comment (finnhodgkin) [inline react-backend/functions/pdfgenerator.js]: Result can just be a `const` defined down here.

Comment (finnhodgkin) [reply react-backend/functions/pdfgenerator.js]: I think the whole thing can be

```js
  const printer = new PdfPrinter(fontDescriptors);
  const doc = printer.createPdfKitDocument(docDefinition);

  const chunks = [];

  doc.on('data', chunk => {
    chunks.push(chunk);
  });

  doc.on('end', () => {
    resolve(Buffer.concat(chunks).toString('base64'));
  });
  doc.end();
```

File: react-backend/client/src/components/admin/adminCBSettingsPage.js [hunk 2]
```diff
@@ -1,8 +1,12 @@
 import React, { Component } from 'react';
+import Dropzone from 'react-dropzone';
 import { Link } from 'react-router-dom';
 import { Button } from '../visitors/button';
 import { Logoutbutton } from '../visitors/logoutbutton';
 
+const CLOUDINARY_UPLOAD_PRESET = 'cklrrn9k';
```
[the remark this answers is not in the document]
Comment (azayneeva) [reply react-backend/client/src/components/admin/adminCBSettingsPage.js]: sure :)

File: react-backend/client/src/components/admin/adminCBSettingsPage.js [hunk 3]
```diff
@@ -283,8 +329,7 @@ export class AdminCBSettingsPage extends Component {
   };
 
   render() {
-    return this.state.reauthenticated
-      ? this.renderAuthenticated()
-      : this.renderNotAuthenticated();
+    console.log(this.state.cbLogo);
```
[the remark this answers is not in the document]
Comment (azayneeva) [reply react-backend/client/src/components/admin/adminCBSettingsPage.js]: 👍

File: react-backend/database/queries/CBqueries/getCBFromEmail.js [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
       .query(getCBQuery, [email])
-      .then((res) => {
+      .then(res => {
         if (res.rowCount) {
           return resolve(res.rows[0]);
         }
         reject('no results found');
       })
-      .catch((err) => {
+      .catch(err => {
```
[the remark this answers is not in the document]
Comment (azayneeva) [reply react-backend/database/queries/CBqueries/getCBFromEmail.js]: done

File: react-backend/functions/pdfgenerator.js (hunk 1 above)
Comment (azayneeva) [reply react-backend/functions/pdfgenerator.js]: done :)

File: react-backend/client/src/components/admin/adminCBSettingsPage.js [hunk 5]
```diff
@@ -238,11 +279,18 @@ export class AdminCBSettingsPage extends Component {
               value={this.state.email}
             />
           </label>
-          <button className="Button" onClick={submitHandler}>
-            Submit
-          </button>
         </form>
-
+        <div className="Dropzone">
```
[the remark this answers is not in the document]
Comment (azayneeva) [reply react-backend/client/src/components/admin/adminCBSettingsPage.js]: Done! :)

**Pull request merged**